discovery-infra/tests/test_kube_api.py

```diff
@@ -129,13 +130,17 @@ def kube_api_test(kube_api_context, nodes, cluster_config, proxy_server=None, *,
     nodes.start_all()
 
     logger.info('waiting for host agent')
-    for agent in cluster_deployment.wait_for_agents(len(nodes)):
+    agents = cluster_deployment.wait_for_agents(len(nodes))
+    for agent in agents:
         agent.approve()
         set_agent_hostname(nodes[0], agent, is_ipv4)  # Currently only supports single node
 
     if len(nodes) == 1:
         set_single_node_ip(cluster_deployment, nodes, is_ipv4)
 
+    logger.info("Waiting for agent status verification")
+    Agent.wait_for_agents_to_install(agents, len(nodes))
```

Comment:
in this stage ``len(agents)`` and ``len(nodes)`` is the same, no?
why do we need them both?